plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -159,6 +178,49 @@ public class KVMStorageProcessor implements StorageProcessor {
      */
     private long waitDelayForVirshCommands = 1000l;
 
+    private int incrementalSnapshotTimeout;
+
+    private static final String CHECKPOINT_XML_TEMP_DIR = "/tmp/cloudstack/checkpointXMLs";
+
+    private static final String BACKUP_XML_TEMP_DIR = "/tmp/cloudstack/backupXMLs";
+
+    private static final String BACKUP_BEGIN_COMMAND = "virsh backup-begin --domain %s --backupxml %s --checkpointxml %s";
+
+    private static final String BACKUP_XML = "<domainbackup><disks><disk name='%s' type='file'><target file='%s'/><driver type='qcow2'/></disk></disks></domainbackup>";
+
+    private static final String INCREMENTAL_BACKUP_XML = "<domainbackup><incremental>%s</incremental><disks><disk name='%s' type='file'><target file='%s'/><driver type='qcow2'/></disk></disks></domainbackup>";
+
+    private static final String CHECKPOINT_XML = "<domaincheckpoint><name>%s</name><disks><disk name='%s' checkpoint='bitmap'/></disks></domaincheckpoint>";
+
+    private static final String CHECKPOINT_DUMP_XML_COMMAND = "virsh checkpoint-dumpxml --domain %s --checkpointname %s --no-domain";
+
+    private static final String DOMJOBINFO_COMPLETED_COMMAND = "virsh domjobinfo --domain %s --completed";
+
+    private static final String DOMJOBABORT_COMMAND = "virsh domjobabort --domain %s";
+
+    private static String CHECKPOINT_DELETE_COMMAND = "virsh checkpoint-delete --domain %s --checkpointname %s";
+
+    private static final String DUMMY_VM_XML = "<domain type='qemu'>\n" +
```

Comment:
@JoaoJandre, I'm using Rocky Linux 8.9 with libvirt version 8.0 and qemu 6.2.
When I try to take a snapshot on a stopped VM I've got
`error: Cannot check QEMU binary /usr/bin/qemu-system-x86_64: No such file or directory`
I made a copy of `usr/libexec/qemu-kvm` to the `/usr/bin/qemu-system-x86_64` and after another try I've got this
`error: unsupported configuration: Emulator '/usr/bin/qemu-system-x86_64' does not support machine type 'pc-i440fx-2.9'`
probably for machine type you should set something universal or none and for the emulator to set the right path depending on the distro

Reply:
@slavkap This should be fixed now, but I do not have a Rocky Linux env to test it, could you verify if it is working now?

Reply:
thanks, @JoaoJandre, all looks good! I don't know if the emulator should be mandatory. Here is my dummy XML:

```
<domain type='qemu'>
  <name>DUMMY-VM-23ae60aa-be86-4a7a-a544-0315c1b3d4b4</name>
  <memory unit='MiB'>256</memory>
  <currentMemory unit='MiB'>256</currentMemory>
  <vcpu>1</vcpu>
  <os>
    <type arch='x86_64' machine='pc'>hvm</type>
    <boot dev='hd'/>
  </os>
  <devices>
    <emulator></emulator>
    <disk type='file' device='disk'>
      <driver name='qemu' type='qcow2' cache='none'/>
      <source file='/mnt/b206833c-4a77-3b14-9de5-3ac91cc6c417/0bfab443-f9a3-4c67-9fd2-6fc77e8afa60'/>
      <target dev='sda'/>
    </disk>
    <graphics type='vnc' port='-1'/>
  </devices>
</domain>
```

Reply:
@slavkap I had some issues when the emulator was not informed. Since `LibvirtComputingResource` has the path for it, I don't see any issue in using it. We also always use it when creating user/system VMs.